src/services/alpacaService.ts

```diff
@@ -112,12 +112,15 @@ export const takeProfitSellOrder = async (symbol: string) => {
     throw Error;
   }
 
-  const value = await getHoldingInTicker(symbol);
+  // sell ~50%, floored value to prevent fractional trades.
+  const quantity = Math.floor(totalQuantity * 0.5);
```

Comment:
Perhaps do Math.ceil instead? Otherwise we might never sell everything (the .floor results in "0", which most likely would make alpaca API return error).